src/components/nav-mesh-helper.js

```diff
@@ -0,0 +1,16 @@
+AFRAME.registerComponent("nav-mesh-helper", {
+  schema: {
+    teleportControls: { type: "selectorAll", default: "[teleport-controls]" }
+  },
+
+  init: function() {
+    const teleportControls = this.data.teleportControls;
+    this.el.addEventListener("model-loaded", () => {
```

Comment:
I think this gets called multiple times if the environment has multiple layers. Not sure if you want to debounce queryCollisionEntities, if it's expensive.

Reply:
Looks like we have a `bundleloaded` event fired by `gltf-bundle` after all the `model-loaded` events have fired